README.md

```diff
@@ -283,6 +286,72 @@ const NewComponent = exclude(Component);
 
 Stops `prepare` traversal at `Component`. Useful for optimizing the `prepare` traversal to visit the minimum number of nodes.
 
+#### useService
+
+*React Hooks were introduced in React v16.8. Make sure you are using a compatible version.*
+
+```js
+import {useService} from 'fusion-react';
+import {ExampleToken} from 'fusion-tokens';
+
+function Component() {
+  const service = useService(ExampleToken);
+  return (
+    <button onClick={service}>Invoke Service</button>
+  );
+}
+```
+
+- `token: Token<TService>` - Required. The token used to look up the registered plugin that resolves to `TService`.
+- `service: TService` - The service provided by the registered plugin.
+
+If no plugin has been registered to this token, an exception is thrown.
+
+#### ServiceConsumer
+
+```js
+import {ServiceConsumer} from 'fusion-react';
+import {ExampleToken} from 'fusion-tokens';
+
+function Component() {
+  return (
+    <ServiceConsumer token={ExampleToken}>
+      {service => (
+        <button onClick={service}>Invoke Service</button>
+      )}
+    </ServiceConsumer>
+  );
+}
+```
+
+- `token: Token` - Required. The token used to lookup the registered plugin.
+- `children: Service => React.Element<any>` - Required. Render prop that is passed the registered service. Should return the React Element to render.
+- `service: any` - The service provided by the registered plugin.
+
+This is the same pattern as the `useService` hook. Opt for using the hook. `ServiceConsumer` is provided as a replacement for any legacy Context usage that may exist.
+
+#### FusionContext
+
+```js
+import {useContext} from 'react';
+import {FusionContext} from 'fusion-react';
+
+function Component() {
+  const ctx = useContext(FusionContext);
```

Comment:
I've been unable to find any client-side plugins that use this pattern. Is this even useful to have in the browser?

Reply:
we might need to do some refactoring, but I definitely think it will be useful. For one example, web-rpc will definitely benefit. Also potentially apollo client